sdk/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/Hash.scala

```diff
@@ -329,19 +436,26 @@ object Hash {
       templateId: Ref.Identifier,
       arg: Value,
       packageName: Ref.PackageName = Ref.PackageName.assertFromString("default"),
-  ): Hash =
-    builder(Purpose.ContractInstance, aCid2Bytes)
-      .add(packageName)
-      .addIdentifier(templateId)
-      .addTypedValue(arg)
-      .build
+      langVersion: TransactionVersion,
+  ): Hash = {
+    import Ordering.Implicits._
+    val b = builder(
+      Purpose.ContractInstance,
+      aCid2Bytes,
+      langVersion >= TransactionVersion.minContractKeys,
```

Comment:
This looks dangerous in combination with the changes to `SerializableRawContractInstance`. As discussed with @matthiasS-da and @simonmaxen-da, Canton is going to re-version the shipped input contract instances according to the transaction version of the node (or any of those in a view). So with this comparison, if the re-versioning jumps of the min-contract-keys boundary, the hash will change and break Canton in various ways.